x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/preprocessing/Multi.java

```diff
@@ -0,0 +1,234 @@
+        this.inputFields = inputFields.toArray(String[]::new);
+    }
+
+    public Multi(StreamInput in) throws IOException {
+        this.processors = in.readNamedWriteableList(PreProcessor.class).toArray(PreProcessor[]::new);
+        this.custom = in.readBoolean();
+        this.outputFields = in.readOrderedMap(StreamInput::readString, StreamInput::readString);
+        this.inputFields = in.readStringArray();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteableList(Arrays.asList(processors));
+        out.writeBoolean(custom);
+        out.writeMap(outputFields, StreamOutput::writeString, StreamOutput::writeString);
+        out.writeStringArray(inputFields);
+    }
+
+    @Override
+    public String toString() {
+        return Strings.toString(this);
+    }
+
+    @Override
+    public List<String> inputFields() {
+        return Arrays.asList(inputFields);
+    }
+
+    @Override
+    public List<String> outputFields() {
+        return new ArrayList<>(outputFields.keySet());
+    }
+
+    @Override
+    public void process(Map<String, Object> fields) {
+        for (PreProcessor processor : processors) {
+            processor.process(fields);
+        }
+    }
+
+    @Override
+    public Map<String, String> reverseLookup() {
+        // TODO
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getOutputFieldType(String outputField) {
+        return outputFields.get(outputField);
+    }
+
+    @Override
+    public long ramBytesUsed() {
+        long size = SHALLOW_SIZE;
+        size += RamUsageEstimator.sizeOf(processors);
+        size += RamUsageEstimator.sizeOf(inputFields);
+        size += RamUsageEstimator.sizeOfMap(outputFields, 0);
+        return size;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public String getName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        NamedXContentObjectHelper.writeNamedObjects(builder, params, true, PROCESSORS.getPreferredName(), Arrays.asList(processors));
+        builder.field(CUSTOM.getPreferredName(), custom);
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean isCustom() {
+        return custom;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Multi multi = (Multi) o;
+        return Arrays.equals(multi.processors, processors) && custom == multi.custom;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(custom);
+        result = 31 * result + Arrays.hashCode(processors);
+        return result;
```

Comment:
This can be simplified to `return Objects.hash(custom, Arrays.hashCode(processors));`.